test/common/src/main/kotlin/org/codice/compliance/verification/core/CoreVerifier.kt

```diff
@@ -15,17 +15,51 @@ package org.codice.compliance.verification.core
 
 import org.apache.cxf.rs.security.saml.sso.SSOConstants.SIGNATURE
 import org.codice.compliance.SAMLComplianceException
+import org.codice.compliance.SAMLCore_3_2_1_d
 import org.codice.compliance.SAMLCore_5_4_1
 import org.codice.compliance.SAMLCore_5_4_2_a
 import org.codice.compliance.SAMLCore_5_4_2_b
 import org.codice.compliance.SAMLCore_5_4_2_b1
 import org.codice.compliance.SAMLCore_6_1_b
+import org.codice.compliance.SAMLSpecRefMessage
 import org.codice.compliance.allChildren
 import org.codice.compliance.children
 import org.codice.compliance.utils.TestCommon.Companion.ELEMENT
+import org.codice.compliance.utils.TestCommon.Companion.REQUESTER
 import org.w3c.dom.Node
 
 class CoreVerifier(val node: Node) {
+
+    /**
+     * Verifies that a response has the expected status code.
+     * This should be called explicitly if an error is expected.
+     *
+     * @param samlErrorCode - The error code you wish to use for the SAMLComplianceException
+     * @param expectedStatusCode - the uri of the expected status code.
+     * For example, urn:oasis:names:tc:SAML:2.0:status:Requester.
+     */
+    fun verifyErrorStatusCode(samlErrorCode: SAMLSpecRefMessage, expectedStatusCode: String) {
+        val status = node.children("Status")
+        if (status.size != 1)
+            throw SAMLComplianceException.createWithPropertyMessage(samlErrorCode,
+                    property = "Status",
+                    actual = "[]",
+                    expected = "<Status>",
+                    node = node)
+        status[0].children("StatusCode").forEach {
+            val code = it.attributes.getNamedItem("Value").textContent
+            if (code != expectedStatusCode) {
+                var exceptions = arrayOf(samlErrorCode)
+                if (expectedStatusCode == REQUESTER)
+                    exceptions = arrayOf(samlErrorCode, SAMLCore_3_2_1_d)
+                throw SAMLComplianceException.createWithPropertyMessage(*exceptions,
+                        property = "Status Code",
+                        actual = code,
+                        expected = expectedStatusCode,
+                        node = status[0])
+            }
+        }
+    }
```

Comment:
I'm not going to lie, this was not what I was thinking. Was thinking we'd have a separate convenience method for the requester with it's own special error message logic. I think I like this better 😄